sympy/polys/tests/test_polytools.py

```diff
@@ -3933,3 +3937,215 @@ def test_issue_20985():
+                            {x - 1, y**2 - y + 1}, {x + 1, y + 1}, {x + 1, y**2 - y + 1}])
+
+    assert_equal_unordered(factor_system([(x - 1)*(y - 2), (y  - 2)*(z - 3)]),
+                           [{y - 2}, {z - 3, x - 1}])
+    assert_equal_unordered(factor_system([sin(x) ** 2 + cos(x) ** 2 - 1, x]), [{sin(x) ** 2 + cos(x) ** 2 - 1, x}])
+
+    assert factor_system([sin(x)**2 + cos(x)**2 - 1]) == [{sin(x)**2 + cos(x)**2 - 1}]
+    assert factor_system([sin(x) ** 2 + cos(x) ** 2]) == [{sin(x) ** 2 + cos(x) ** 2}]
+
+    # Generators not symbols
+    expr1 = cos(x) ** 2 - sin(x) ** 2
+    expr2 = cos(x) ** 2 + sin(x) ** 2 - 1
+    result = factor_system([expr1, expr2])
+    expected = [{sin(x)**2 + cos(x)**2 - 1, sin(x) + cos(x)}, {-sin(x) + cos(x), sin(x)**2 + cos(x)**2 - 1}]
+    assert_equal_unordered(result, expected)
+
+    expr3 = (cos(x) + sin(x)) ** 2 - 1
+    expr4 = cos(x) ** 2 - sin(x) ** 2 - cos(2 * x)
+    result = factor_system([expr3, expr4])
+    expected = [{sin(x) + cos(x) - 1, sin(x)**2 - cos(x)**2 + cos(2*x)}, {sin(x) + cos(x) + 1, sin(x)**2 - cos(x)**2 + cos(2*x)}]
+    assert_equal_unordered(result, expected)
+
+    expr5 = (cos(x) + sin(x)) * exp(y) - 1
+    expr6 = (cos(x) - sin(x)) * exp(y) - 1
+    result = factor_system([expr5, expr6])
+    expected = [{-exp(y)*sin(x) + exp(y)*cos(x) - 1, exp(y)*sin(x) + exp(y)*cos(x) - 1}]
+    assert_equal_unordered(result, expected)
+
+
+def _poly_pair_to_tuple(pair):
+    poly, conditions = pair
+    poly_tuple = (str(poly.rep), str(poly.domain), poly.gens)
+    conditions_tuple = tuple(str(cond) for cond in conditions)
+    return (poly_tuple, conditions_tuple)
+
+
+def _expr_pair_to_tuple(pair):
+    expr, condition = pair
+    expr_tuple = (str(expr), str(condition))
+    return expr_tuple
+
+def test_factor_system_poly():
+
+    def poly_pair_to_tuple(pair):
+        poly, conditions = pair
+        poly_tuple = (str(poly.rep), str(poly.domain), poly.gens)
+        return poly_tuple
+
+    def assert_equal_factor_systems(actual, expected):
+        assert len(actual) == len(expected)
+
+        def system_to_hashable(system):
+            # Converting each combo to a hashable set of tuples for comparison
+            return frozenset(
+                frozenset(poly_pair_to_tuple(pair) for pair in combo)
+                for combo in system
+            )
+
+        actual_set = system_to_hashable(actual[0])
+        expected_set = system_to_hashable(expected[0])
+        assert actual_set == expected_set
+
+    p1 = Poly(x ** 2 - 1, x)
+    p2 = Poly(x ** 2 - 4, x)
+    result = factor_system_poly([p1, p2])
+    expected = ([{(Poly(x - 1, x, domain='ZZ'), ()), (Poly(x - 2, x, domain='ZZ'), ())},
+                 {(Poly(x + 2, x, domain='ZZ'), ()), (Poly(x - 1, x, domain='ZZ'), ())},
+                 {(Poly(x + 1, x, domain='ZZ'), ()), (Poly(x - 2, x, domain='ZZ'), ())},
+                 {(Poly(x + 2, x, domain='ZZ'), ()), (Poly(x + 1, x, domain='ZZ'), ())}], [])
```

Comment:
It might be worth using gcd to eliminate/simply redundant systems with univariate polynomials like this.